evalml/pipelines/pipeline_meta.py

```diff
@@ -14,20 +14,23 @@ def check_for_fit(cls, method):
         """
 
         @wraps(method)
-        def _check_for_fit(self, X=None, objective=None):
+        def _check_for_fit(self, *args, **kwargs):
             klass = type(self).__name__
             if not self._is_fitted:
                 raise PipelineNotYetFittedError(
                     f"This {klass} is not fitted yet. You must fit {klass} before calling {method.__name__}."
                 )
-            if method.__name__ == "predict_proba":
```

Comment:
Because we want to support multiple methods now with different signatures (ex: predict takes X, objective but transform takes X, y), we need to make this more general.

Reply:
Interesting question! I think the difficulty is with the signature of `_check_for_fit` not being the most flexible since that's wrapped around all of our methods. I didn't see any downsides of being more general--if the user passes in the wrong number of arguments, the exception thrown is pretty clear--so I think this was mostly an implementation detail and a nice cleanup! Open to hearing concerns though 😅